spring-state-machine/src/test/java/com/baeldung/spring/statemachine/StateMachineTest.java

```diff
@@ -0,0 +1,53 @@
+package com.baeldung.spring.stateMachine;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.statemachine.StateMachine;
+
+import com.baeldung.spring.stateMachine.config.SimpleStateMachineConfiguration;
+import com.jayway.awaitility.Awaitility;
+
+public class StateMachineTest {
+
+    private StateMachine stateMachine;
+
+    @Before
+    public void setUp() {
+        AnnotationConfigApplicationContext ctx = new AnnotationConfigApplicationContext(SimpleStateMachineConfiguration.class);
+        stateMachine = ctx.getBean(StateMachine.class);
+        stateMachine.start();
+    }
+
+    @Test
+    public void whenSimpleStringStateMachineEvents_thenEndState() {
+        assertEquals("SI", stateMachine.getState().getId());
+
+        stateMachine.sendEvent("E1");
+        assertEquals("S1", stateMachine.getState().getId());
+
+        stateMachine.sendEvent("E2");
+        assertEquals("S2", stateMachine.getState().getId());
+
+        stateMachine.sendEvent("end");
+        assertEquals("SF", stateMachine.getState().getId());
+
+    }
+
+    @Test
+    public void whenSimpleStringMachineActionState_thenActionExecuted() {
+        stateMachine.sendEvent("E3");
+        assertEquals("S3", stateMachine.getState().getId());
+
+        boolean acceptedE4 = stateMachine.sendEvent("E4");
+
+        assertTrue(acceptedE4);
+        Awaitility.await().atMost(5, TimeUnit.SECONDS).until(() -> assertEquals("S4", stateMachine.getState().getId()));
```

Comment:
Those two assertions should be replaced with the await() call. No need for wrapping the assertion